v2/cmd/wails/internal/commands/dev/dev.go

```diff
@@ -438,12 +450,32 @@ func runFrontendDevWatcherCommand(cwd string, devCommand string) (func(), error)
 		wg.Done()
 	}()
 
+	var viteServerURL string
+	if discoverViteServerURL {
+		buf := bufio.NewReader(stdo)
+		for {
+			line, _, _ := buf.ReadLine()
+			match := bytes.Index(line, []byte("Local:"))
+			if match != -1 {
+				line := strings.TrimSpace(string(line[match+6:]))
+				viteServerURL = stripansi.Strip(line)
+				LogGreen("Vite Server URL: %s", viteServerURL)
+				_, err := url.Parse(viteServerURL)
+				if err != nil {
+					LogRed(err.Error())
+					break
+				}
+				break
+			}
+		}
+	}
+
```

Comment:
According to the documentation of [stdout pipe](https://pkg.go.dev/os/exec#Cmd.StdoutPipe) one should call `wait` after all reads from the pipe have been completed.
So I would assume we shouldn't start the go routine which waits before the search has been finished.